File: mevzuat/update_nav_footer.py
import os
import re

base_dir = '.'
root_index_path = '../index.html'

with open(root_index_path, 'r', encoding='utf-8') as f:
    root_html = f.read()

nav_match = re.search(r'(<nav class="navbar">.*?</nav>)', root_html, flags=re.DOTALL)
footer_match = re.search(r'(<footer class="main-footer">.*?</footer>)', root_html, flags=re.DOTALL)

if not nav_match or not footer_match:
    print("Could not find nav or footer in root index.html")
    exit(1)

nav_html = nav_match.group(1)
footer_html = footer_match.group(1)

# Fix paths for mevzuat subdirectories (from ./ to ../../, from mevzuat/ to ../)
nav_html = nav_html.replace('href="./"', 'href="../../"')
nav_html = nav_html.replace('src="./content/', 'src="../../content/')
nav_html = nav_html.replace('data-target="#products-page"', 'href="../../?products"')
nav_html = nav_html.replace('data-target="#checkout-page"', 'href="../../?checkout"')
nav_html = nav_html.replace('data-target="#login-page"', 'href="../../?login"')
nav_html = nav_html.replace('data-target="#dashboard-page"', 'href="../../?dashboard"')
# Make lists clickable as links in nav
nav_html = nav_html.replace('<li data-target="#products-page" class="active">Ürünlerimiz</li>', '<li><a href="../../?products" style="text-decoration: none; color: inherit;">Ürünlerimiz</a></li>')
nav_html = nav_html.replace('<li data-target="#checkout-page"', '<li ')
nav_html = nav_html.replace('<i class="fa-solid fa-cart-shopping" style="font-size: 1.2rem;"></i> Sepet <span id="cart-badge">0</span></li>', '<a href="../../?checkout" style="text-decoration: none; color: inherit; display: flex; align-items: center; gap: 6px;"><i class="fa-solid fa-cart-shopping" style="font-size: 1.2rem;"></i> Sepet <span id="cart-badge">0</span></a></li>')
nav_html = nav_html.replace('<li id="nav-login-btn"', '<li id="nav-login-btn"')
nav_html = nav_html.replace('<i class="fa-regular fa-user" style="font-size: 1.2rem;"></i> Giriş Yap\n                </li>', '<a href="../../?login" style="text-decoration: none; color: inherit; display: flex; align-items: center; gap: 8px;"><i class="fa-regular fa-user" style="font-size: 1.2rem;"></i> Giriş Yap</a>\n                </li>')
nav_html = nav_html.replace('<i class="fa-solid fa-circle-user" style="font-size: 1.3rem;"></i> Profil\n                </li>', '<a href="../../?dashboard" style="text-decoration: none; color: inherit; display: flex; align-items: center; gap: 8px;"><i class="fa-solid fa-circle-user" style="font-size: 1.3rem;"></i> Profil</a>\n                </li>')

# Remove class="nav-trigger" as we use standard links
nav_html = nav_html.replace('class="nav-trigger"', '')

footer_html = footer_html.replace('href="mevzuat/', 'href="../')

for d in os.listdir(base_dir):
    dir_path = os.path.join(base_dir, d)
    if os.path.isdir(dir_path):
        file_path = os.path.join(dir_path, 'index.html')
        if os.path.exists(file_path):
            with open(file_path, 'r', encoding='utf-8') as f:
                content = f.read()
            
            # replace nav
            content = re.sub(r'<nav class="navbar">.*?</nav>', nav_html, content, flags=re.DOTALL)
            
            # replace footer
            content = re.sub(r'<footer class="main-footer">.*?</footer>', footer_html, content, flags=re.DOTALL)
            
            # replace inline script with mevzuat_script.js
            content = re.sub(r'<script>\s*\$\(document\)\.ready\(function\(\) \{.*?\updateCartBadge\(\);\s*\}\);\s*</script>', '<script type="module" src="../mevzuat_script.js"></script>', content, flags=re.DOTALL)
            
            with open(file_path, 'w', encoding='utf-8') as f:
                f.write(content)
            print(f'Updated {file_path}')

File: mevzuat/update_nav_footer.js
const fs = require('fs');
const path = require('path');

const baseDir = '.';
const rootIndexPath = '../index.html';

const rootHtml = fs.readFileSync(rootIndexPath, 'utf-8');

const navMatch = rootHtml.match(/(<nav class="navbar">[\s\S]*?<\/nav>)/);
const footerMatch = rootHtml.match(/(<footer class="main-footer">[\s\S]*?<\/footer>)/);

if (!navMatch || !footerMatch) {
    console.error("Could not find nav or footer in root index.html");
    process.exit(1);
}

let navHtml = navMatch[1];
let footerHtml = footerMatch[1];

// Mobil alt menü + stilleri ve güncel myStyle.css sürümü de ana sayfadan kopyalanır.
// Alt menüdeki tek-sayfa (SPA) hedefleri burada normal sayfa geçişine çevrilir; aktif sekme olmaz.
const bottomNavMatch = rootHtml.match(/<!-- MOBILE BOTTOM NAV -->\s*<div class="bottom-nav-bar"[\s\S]*?\n    <\/div>/);
const bottomStyleMatch = rootHtml.match(/<!-- Sayfaya ozel mobil alt menu stilleri[\s\S]*?<\/style>/);
const cssVersion = (rootHtml.match(/myStyle\.css\?v=(\d+)/) || [])[1];
if (!bottomNavMatch || !bottomStyleMatch || !cssVersion) {
    console.error("Could not find bottom nav, its styles or myStyle.css version in root index.html");
    process.exit(1);
}
const bottomNavHtml = bottomNavMatch[0]
    .replace('class="bottom-nav-item active"', 'class="bottom-nav-item"')
    .replace(/data-target="#(products|checkout|login|dashboard)-page"/g, `onclick="location.href='../../?$1'"`);
const bottomStyleHtml = bottomStyleMatch[0];

navHtml = navHtml.replace(/href="\.\/"/g, 'href="../../"');
navHtml = navHtml.replace(/src="\.\/content\//g, 'src="../../content/');
navHtml = navHtml.replace(/data-target="#products-page"/g, 'href="../../?products"');
navHtml = navHtml.replace(/data-target="#checkout-page"/g, 'href="../../?checkout"');
navHtml = navHtml.replace(/data-target="#login-page"/g, 'href="../../?login"');
navHtml = navHtml.replace(/data-target="#dashboard-page"/g, 'href="../../?dashboard"');
navHtml = navHtml.replace('<li href="../../?products" class="active">Ürünlerimiz</li>', '<li><a href="../../?products" style="text-decoration: none; color: inherit;">Ürünlerimiz</a></li>');
navHtml = navHtml.replace('<li href="../../?checkout"', '<li ');
navHtml = navHtml.replace('<i class="fa-solid fa-cart-shopping" style="font-size: 1.2rem;"></i> Sepet <span id="cart-badge">0</span></li>', '<a href="../../?checkout" style="text-decoration: none; color: inherit; display: flex; align-items: center; gap: 6px;"><i class="fa-solid fa-cart-shopping" style="font-size: 1.2rem;"></i> Sepet <span id="cart-badge">0</span></a></li>');
navHtml = navHtml.replace('<li id="nav-login-btn"', '<li id="nav-login-btn"');
navHtml = navHtml.replace('<i class="fa-regular fa-user" style="font-size: 1.2rem;"></i> Giriş Yap\n                </li>', '<a href="../../?login" style="text-decoration: none; color: inherit; display: flex; align-items: center; gap: 8px;"><i class="fa-regular fa-user" style="font-size: 1.2rem;"></i> Giriş Yap</a>\n                </li>');
navHtml = navHtml.replace('<i class="fa-solid fa-circle-user" style="font-size: 1.3rem;"></i> Profil\n                </li>', '<a href="../../?dashboard" style="text-decoration: none; color: inherit; display: flex; align-items: center; gap: 8px;"><i class="fa-solid fa-circle-user" style="font-size: 1.3rem;"></i> Profil</a>\n                </li>');
navHtml = navHtml.replace(/<div class="brand-logo[^>]*>([\s\S]*?)<\/div>/, '<div class="brand-logo" style="cursor: pointer;"><a href="../../?products" style="text-decoration: none; color: inherit;">$1</a></div>');
navHtml = navHtml.replace(/class="nav-trigger"/g, '');

footerHtml = footerHtml.replace(/href="mevzuat\//g, 'href="../');

const dirs = fs.readdirSync(baseDir);

for (const d of dirs) {
    const dirPath = path.join(baseDir, d);
    if (fs.statSync(dirPath).isDirectory()) {
        const filePath = path.join(dirPath, 'index.html');
        if (fs.existsSync(filePath)) {
            let content = fs.readFileSync(filePath, 'utf-8');
            
            content = content.replace(/<nav class="navbar">[\s\S]*?<\/nav>/, navHtml);
            content = content.replace(/<footer class="main-footer">[\s\S]*?<\/footer>/, footerHtml);
            content = content.replace(/myStyle\.css\?v=\d+/, `myStyle.css?v=${cssVersion}`);
            const styleRe = /<!-- Sayfaya ozel mobil alt menu stilleri[\s\S]*?<\/style>/;
            content = styleRe.test(content) ? content.replace(styleRe, () => bottomStyleHtml) : content.replace('</head>', () => `    ${bottomStyleHtml}\n</head>`);
            const navRe = /<!-- MOBILE BOTTOM NAV -->\s*<div class="bottom-nav-bar"[\s\S]*?\n    <\/div>/;
            content = navRe.test(content) ? content.replace(navRe, () => bottomNavHtml) : content.replace('</body>', () => `    ${bottomNavHtml}\n</body>`);
            content = content.replace(/<script>\s*\$\(document\)\.ready\(function\(\) \{[\s\S]*?updateCartBadge\(\);\s*\}\);\s*<\/script>/, '<script type="module" src="../mevzuat_script.js?v=2"></script>');
            content = content.replace(/mevzuat_script\.js(\?v=\d+)?"/, 'mevzuat_script.js?v=2"');
            
            fs.writeFileSync(filePath, content, 'utf-8');
            console.log(`Updated ${filePath}`);
        }
    }
}
